R/load_runner.R

```diff
@@ -45,22 +50,26 @@ run_load <- function(con, load, extracted_data, transformed_data, test_queries,
 
 do_load <- function(con, load, extracted_data, transformed_data, path,
                     test_file, test_queries, pre_load, post_load, log_table,
-                    log_data, dry_run) {
-  message("Running load:")
+                    log_data, transaction, dry_run) {
+  message(
+    sprintf("Running load %s:",
+            transaction %?% "in a transaction" %:% "not in a transaction"))
   message("\t- Running test queries before making any changes")
-  before <- test_queries(con)
-  if (!is.null(pre_load)) {
-    message("\t- Running pre-load")
-    pre_load(transformed_data, con)
-  }
-  message("\t- Running load step")
-  load(transformed_data, con)
-  if (!is.null(post_load)) {
-    message("\t- Running post-load")
-    post_load(transformed_data, con)
-  }
-  message("\t- Running test queries after making changes")
-  after <- test_queries(con)
+  withr::with_dir(path, {
+    before <- test_queries(con)
+    if (!is.null(pre_load)) {
+      message("\t- Running pre-load")
+      pre_load(transformed_data, con)
+    }
+    message("\t- Running load step")
+    load(transformed_data, con)
+    if (!is.null(post_load)) {
+      message("\t- Running post-load")
+      post_load(transformed_data, con)
+    }
+    message("\t- Running test queries after making changes")
+    after <- test_queries(con)
+  })
   test_path <- file.path(path, test_file)
```

Comment:
you could roll this section into the changed-dir above I think

Reply:
Moved this into the withr block too - I agree we should be consistent about wd